pretrain/redpajama.py

```diff
@@ -68,6 +67,7 @@ def setup(
     val_data_dir: Optional[Path] = None,
     precision: Optional[str] = None,
     tpu: bool = False,
+    resume: bool | Path = False,
```

Comment:
Not using Union will break jsonargparse without `from __future__ import annotations`. I suggest you use the old style and if you'd like, update all annotations to use the new style in a new PR

Reply:
Oh it was working for me. I can use Union for maybe older versions